adapters/handlers/rest/configure_api.go

```diff
@@ -374,6 +375,35 @@ func MakeAppState(ctx context.Context, options *swag.CommandLineOptionsGroup) *s
 		remoteIndexClient, appState.Logger, appState.ServerConfig.Config.Persistence.DataPath)
 	appState.Scaler = scaler
 
+	// let classTenantDataEvents be nil if the metadata server is not enabled since the metadata
+	// server/querierManager are the users of the channel
+	var classTenantDataEvents chan metadataserver.ClassTenant
+	querierManager := metadataserver.NewQuerierManager(appState.Logger)
+	metadataServer := metadataserver.NewServer(
```

Comment:
Can we move that whole block of code under a config option ? Ideally I want to experimental code being executed *at all* unless we have enabled it.

Reply:
Agree. +1